File: 876. Middle of the Linked List/go/middle_of_the_linked_list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 
 func middleNode(head *ListNode) *ListNode {
    chs := head
    chf := head
    for chf != nil && chf.Next != nil {
        chf = chf.Next.Next
        chs = chs.Next
    }
    return chs
}

File: 876. Middle of the Linked List/c/middle_of_the_linked_list.c
#include <stdio.h>

// Definition for singly-linked list.
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* middleNode(struct ListNode* head){
    struct ListNode*chf = head;
    struct ListNode*chs = head;
    while (chf && chf->next) {
        chf = chf->next->next;
        chs = chs->next;
    }
    return chs;
}

